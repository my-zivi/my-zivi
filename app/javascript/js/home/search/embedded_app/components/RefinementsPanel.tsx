import { ClearRefinements } from 'react-instantsearch-dom';
import React from 'preact/compat';
import { Collapse } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mobile } from 'is_js';
import { IconName } from '@fortawesome/fontawesome-svg-core';
import {
  CantonDisplayNameRefinement,
  CategoryDisplayNameRefinement,
  LanguageRefinement,
  MinimumServiceMonthsRefinement,
  PriorityProgramRefinement,
} from './refinements';

class RefinementsPanel extends React.Component<Record<string, never>, { isCollapsed: boolean }> {
  constructor() {
    super();
    this.state = { isCollapsed: mobile() };
  }

  private panelContent(): JSX.Element {
    return (
      <div className="mt-3">
        <strong>{MyZivi.translations.search.refinements.language}</strong>
        <LanguageRefinement />
        <hr />
        <strong>{MyZivi.translations.search.refinements.category}</strong>
        <CategoryDisplayNameRefinement />
        <hr />
        <strong>{MyZivi.translations.search.refinements.canton}</strong>
        <CantonDisplayNameRefinement />
        <hr />
        <strong>{MyZivi.translations.search.refinements.minimumServiceMonths}</strong>
        <MinimumServiceMonthsRefinement />
        <hr />
        <strong>{MyZivi.translations.search.refinements.priorityProgram}</strong>
        <PriorityProgramRefinement />
        <div className="d-block d-lg-none mt-2">
          {this.collapseButton('chevron-down', 'chevron-up')}
        </div>
      </div>
    );
  }

  private collapseButton(collapsedIcon: IconName = 'chevron-up', expandedIcon: IconName = 'chevron-down') {
    return (
      <a className="mx-0 w-100 d-block"
         href="#"
         onClick={(e) => {
           e.preventDefault();
           this.setState({ isCollapsed: !this.state.isCollapsed });
         }}
      >
        {this.state.isCollapsed
          ? <FontAwesomeIcon icon={['fas', collapsedIcon]} className="mr-1" />
          : <FontAwesomeIcon icon={['fas', expandedIcon]} className="mr-1" />}
        {MyZivi.translations.search.refinements.collapseButton}
      </a>
    );
  }

  render(): JSX.Element {
    return (
      <>
        <div className="card card-body">
          {this.collapseButton()}
          <Collapse in={!this.state.isCollapsed}>
            {this.panelContent()}
          </Collapse>
        </div>
        <ClearRefinements clearsQuery={false} className="my-3" translations={{
          reset: MyZivi.translations.search.refinements.controls.reset,
        }} />
      </>
    );
  }
}

export default RefinementsPanel;
