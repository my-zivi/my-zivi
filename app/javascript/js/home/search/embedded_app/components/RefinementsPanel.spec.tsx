import React from 'preact/compat';
import { shallow } from 'enzyme';
import RefinementsPanel from './RefinementsPanel';

describe('RefinementsPanel', () => {
  it('renders correctly', () => {
    expect(
      shallow(<RefinementsPanel />),
    ).toMatchSnapshot();
  });
});
