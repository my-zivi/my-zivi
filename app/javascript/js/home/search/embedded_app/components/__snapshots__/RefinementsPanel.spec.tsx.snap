// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefinementsPanel renders correctly 1`] = `
Array [
  <div
    className="card card-body"
  >
    <a
      className="mx-0 w-100 d-block"
      href="#"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        className="mr-1"
        icon={
          Array [
            "fas",
            "chevron-down",
          ]
        }
      />
      string
    </a>
    <ForwardRef()
      appear={false}
      getDimensionValue={[Function]}
      in={true}
      mountOnEnter={false}
      timeout={300}
      unmountOnExit={false}
    >
      <div
        className="mt-3"
      >
        <strong>
          string
        </strong>
        <LanguageRefinement />
        <hr />
        <strong>
          string
        </strong>
        <CategoryDisplayNameRefinement />
        <hr />
        <strong>
          string
        </strong>
        <CantonDisplayNameRefinement />
        <hr />
        <strong>
          string
        </strong>
        <MinimumServiceMonthsRefinement />
        <hr />
        <strong>
          string
        </strong>
        <PriorityProgramRefinement />
        <div
          className="d-block d-lg-none mt-2"
        >
          <a
            className="mx-0 w-100 d-block"
            href="#"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              className="mr-1"
              icon={
                Array [
                  "fas",
                  "chevron-up",
                ]
              }
            />
            string
          </a>
        </div>
      </div>
    </ForwardRef()>
  </div>,
  <ConnectorWrapper
    className="my-3"
    clearsQuery={false}
    translations={
      Object {
        "reset": "string",
      }
    }
  />,
]
`;
